src/main/resources/petriNets/engine-processes/dashboard_item.xml

```diff
@@ -0,0 +1,736 @@
+        <event type="set">
+            <id>preference_items_list_set</id>
+            <actions phase="post">
+                <action>
+                    icon_color: f.icon_color,
+                    font_color: f.font_color,
+                    font_weight: f.font_weight,
+                    inherit_name: f.inherit_name,
+                    inherit_icon: f.inherit_icon,
+                    preference_item_taskRef: f.preference_item_taskRef,
+                    filter_divider: f.filter_divider,
+                    preference_items_list: f.preference_items_list,
+                    trans: t.configuration;
+
+                    change inherit_icon value { true }
+                    change inherit_name value { true }
+                    if (preference_items_list.value != null &amp;&amp; org.bson.types.ObjectId.isValid(preference_items_list.value)) {
+                        make [inherit_name, inherit_icon, icon_color, font_color, font_weight], editable on trans when { true }
+                        make [preference_item_taskRef, filter_divider], visible on trans when { true }
+                        def preferenceItemCase = workflowService.findOne(preference_items_list.value)
+                        change preference_item_taskRef value { [preferenceItemCase.tasks.find { it.transition == "item_settings" }?.task] }
+                    } else {
+                        make [inherit_name, inherit_icon, icon_color, font_color, font_weight], visible on trans when { true }
+                        make [preference_item_taskRef, filter_divider], hidden on trans when { true }
+                        change preference_item_taskRef value { [] }
+                    }
+                </action>
+            </actions>
+        </event>
+    </data>
+    <data type="boolean" immediate="true">
+        <id>is_internal</id>
+        <title name="is_internal">Internal item</title>
+        <desc name="is_internal_desc">Internal menu item or link to external website URL</desc>
+        <init>true</init>
+        <event type="set">
+            <id>is_internal_set</id>
+            <actions phase="post">
+                <action>
+                    item_name: f.item_name,
+                    item_icon: f.item_icon,
+                    external_icon: f.external_icon,
+                    preference_items_list: f.preference_items_list,
+                    external_url: f.external_url,
+                    is_internal: f.is_internal;
+
+                    change external_icon value { false }
+                    if (is_internal.value) {
+                        change preference_items_list value { "" }
+                    } else {
+                        change item_icon value { "" }
+                        change item_name value { new com.netgrif.application.engine.petrinet.domain.I18nString("") }
+                        change external_url value { "" }
+                    }
+                </action>
+                <action>
```

Comment:
Is there any reason to have 2 actions instead of just 1?

Reply:
There is the issue with I18nField, when changing it value and behavior in one action, only one of changes is propagated to FE. Needs to be separated into two actions. Until resolved, this needs to be split into two actions.